File: pl-domain/country-service/src/main/java/com/polylab/countries/domain/service/CountryServiceImpl.java
package com.polylab.countries.domain.service;

import com.jxareas.jxcore.domain.service.CrudDomainService;
import com.polylab.countries.persistence.model.Country;
import com.polylab.countries.persistence.repository.CountryRepository;
import org.springframework.stereotype.Service;

@Service
public class CountryServiceImpl extends CrudDomainService<Country, Integer> implements CountryService {
    public CountryServiceImpl(CountryRepository repository) {
        super(repository);
    }
}

File: pl-domain/country-service/src/main/java/com/polylab/countries/domain/service/CountryServiceImpl.kt
package com.polylab.countries.domain.service

import com.jxareas.jxelerator.domain.service.PersistentService
import com.polylab.countries.persistence.model.Country
import com.polylab.countries.persistence.repository.CountryRepository
import org.springframework.stereotype.Service

@Service
class CountryServiceImpl(repository: CountryRepository) :
    CountryService, PersistentService<Country, Int>(repository)
